### PR TITLE
Ruby on Rails capstone project - Budget app

### PR DESCRIPTION
**Pull request by musangamfure**
Hello Reviewer👋,

In this PR, I have completed all the requirements for the development and production parts of the Budget App capstone project.

**Here is what I have implemented:**

- [ ] Create the following pages:
    - Splash screen
    - Sign-up and login pages
    - Home page (categories page)
    - Transactions page
    - "Add a new category" page
    - "Add a new transaction" page
- [ ] Create unit and integration tests
- [ ] Technical requirements
    - Postgres for the database
    - Devise for authentication
    - Validate user input

- [ ] Deployed the app on render
- [ ] No linter error
- [ ] Gitflow follows

Kindly check my code and let me know what you think

Best Regards

Happy Coding🎉🎉

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -0,0 +1,174 @@
...
+<p align="right">(<a href="#readme-top">back to top</a>)</p>
+
+<!-- LIVE DEMO -->
+
+## 🚀 Live Demo <a name="live-demo"></a>
+
+- [Live Demo Link](https://budgetapp-mu8y.onrender.com/)
+
+- [Presentation Loom Video](https://www.loom.com/share/66c8a10a109f48fa96cf74a2b6ff88a5?sid=e7208321-c4dd-4484-b23b-451e836ba801)
+
```
Comment (Nemwel-Boniface) [inline README.md]: - [x] The project that is presented in this video is clearly different from the project that you currently have. Some of the aspects that you currently have in your current project are not the same as what is here in the presented video.

**This is the landing page of your presented video:**

image

**This is the landing page of the project that is currently in this repo:**

image

**As it is clear that these are two different projects, Kindly do the following:**

- [x] Kindly re-record the video this time round ensuring that you showcase the project that you currently have in this repo.

File: app/models/category.rb [hunk 2]
```diff
@@ -0,0 +1,9 @@
+class Category < ApplicationRecord
+  has_many :category_bills, dependent: :destroy
+  has_many :bills, through: :category_bills
+  belongs_to :user, class_name: 'User', foreign_key: 'user_id'
+
+  has_one_attached :icon
+
+  validates :name, presence: true
+end
```
Comment (Nemwel-Boniface) [inline app/models/category.rb]: - [x] I noticed that in you seem to have missed validating the icon and user_id columns. Kindly include validations for these column as well.

File: app/models/category_bill.rb [hunk 3]
```diff
@@ -0,0 +1,4 @@
+class CategoryBill < ApplicationRecord
+  belongs_to :category
+  belongs_to :bill
+end
```
Comment (Nemwel-Boniface) [inline app/models/category_bill.rb]: - [x]  I noticed that in you seem to have missed validating the category_id and bill_id_columns. Kindly include validations for these column as well.

File: app/models/user.rb [hunk 4]
```diff
@@ -0,0 +1,11 @@
...
+  # Include default devise modules. Others available are:
+  # :confirmable, :lockable, :timeoutable, :trackable and :omniauthable
+  devise :database_authenticatable, :registerable,
+         :recoverable, :rememberable, :validatable
+
+  has_many :categories, class_name: 'Category', dependent: :destroy, foreign_key: 'user_id'
+  has_many :bills, class_name: 'Bill', dependent: :destroy, foreign_key: 'user_id'
+
+  validates :name, presence: true
+end
```
Comment (Nemwel-Boniface) [inline app/models/user.rb]: - [x] KIndly add validations for the email column as well.

**Review by Nemwel-Boniface — CHANGES_REQUESTED** [review]
# Hello @musangamfure  :wave: 

### This is a manual review

# :recycle:  Request Changes :recycle: 

- [ ] You did well in your project. There are however a few changes left to proceed to the next milestone. :rocket: 



**Kindly note the following for future reviews:**
1. For group capstone projects the correct flow for requesting reviews is Github flow which means dev merging to main. (In this case, your project used the wrong flow).
2. For solo capstone projects the flow for requesting reviews is Gitflow which means feature branch merging to dev.


- [ ] Please find below the changes you need to make and once you are ready please tag me @Nemwel-Boniface so that I can get notification. :rocket: 


Cheers and Happy coding! :rocket:

**Comment by musangamfure** [comment]
Hello @Nemwel-Boniface 

Thank you for your professional comments and guidance on the changes I needed to make. I confirm that I have done all you have asked me to change.

Kindly check again and let me know if there is anything else I can improve.

Best

Happy Coding🎉🎉

File: README.md [hunk 5]
```diff
@@ -0,0 +1,174 @@
...
+<p align="right">(<a href="#readme-top">back to top</a>)</p>
+
+<!-- LIVE DEMO -->
+
+## 🚀 Live Demo <a name="live-demo"></a>
+
+- [Live Demo Link](https://budgetapp-mu8y.onrender.com/)
+
+- [Presentation Loom Video](https://www.loom.com/share/1b4f1ef209e94528bf73a1cf2ccb182c?sid=d3588348-3e8c-48ba-8ccf-3057649bc7af)
+
```
Comment (Nemwel-Boniface) [inline README.md]: - [x] While you did well updating the demo video, I was having a challenge hearing what you were saying having in some sections being in audible, Kindly do the following:
   -  KIndly retake the video ensuring that your microphone is clear and that people can hear you clear;y.
   -  KIndly record a video demonstrating the local version of the project and not the deployed version.


Once this is done, I believe your project will be ready for approval. :rocket:

**Review by Nemwel-Boniface — CHANGES_REQUESTED** [review]
# Hello @musangamfure  :wave: 

### This is a manual review

# :recycle:  Request Changes :recycle: 

- [ ] You did well in making the changes I requested There is however a small change left to proceed to the next milestone. :rocket: 




- [ ] Please find below the changes you need to make and once you are ready please tag me @Nemwel-Boniface so that I can get notification. :rocket: 


Cheers and Happy coding! :rocket:

File: README.md (hunk 5 above)
Comment (musangamfure) [reply README.md]: Hello @Nemwel-Boniface 

Thanks for your detailed comments and guidance on what to improve. As per your request, I have redone the loom video.

Plz check and let me know.

**Review by Nemwel-Boniface — APPROVED** [review]
# Hello @musangamfure  :wave: 

### This is a manual review

# :heavy_check_mark:  Approved :heavy_check_mark: 




- [x] Thank you for making all the changes that I had requested earlier. Your project is ready for approval. :rocket: 

Cheers and Happy coding!

**Comment by musangamfure** [comment]
**Hello @Nemwel-Boniface** 

Thank you so much for your kindness and overall professionalism.

**Pull request merged**